core/arch/interp.c

```diff
@@ -3683,6 +3695,12 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
 # endif /* X86 */
 #endif /* WINDOWS */
 
+        if (dcontext->forged_exception_addr == bb->start_pc) {
```

Comment:
I do not understand this one. You are saying that the `break` will not exit the `while (true)` ? Is not that what happens for an invalid instruction ?